src/localization/basestation_gps_driver.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python3
+import rospy
+from serial import Serial
+from pyubx2 import UBXReader, UBX_PROTOCOL, RTCM3_PROTOCOL, protocol
+from rtcm_msgs.msg import Message
+
+
+def main() -> None:
+    rtcm_pub = rospy.Publisher("rtcm", Message, queue_size=1)
+    rospy.init_node("basestation_driver")
+    port = rospy.get_param("basestation_gps_driver/port")
+    baud = rospy.get_param("basestation_gps_driver/baud")
+    svin_started = False
+    svin_complete = False
+
+    # connect to GPS over serial, only read UBX and RTCM messages
+    with Serial(port, baud, timeout=1) as ser:
+        reader = UBXReader(ser, protfilter=(UBX_PROTOCOL | RTCM3_PROTOCOL))
+        while not rospy.is_shutdown():
+            if ser.in_waiting:
+                (raw_msg, msg) = reader.read()
+
+                # skip if message could not be parsed
+                if not msg:
+                    continue
+
+                # publish RTCM messages
+                if protocol(raw_msg) == RTCM3_PROTOCOL:
+                    rtcm_pub.publish(Message(message=raw_msg))
+
+                # print survey-in status
+                elif msg.identity == "NAV-SVIN":
+                    if not svin_started and msg.active:
+                        svin_started = True
+                        rospy.loginfo("basestation survey-in started")
```

Comment:
How often do these print? If a lot I would add a throttle

Reply:
one of them prints at about 10hz which should be fine